src/c++/perf_analyzer/load_worker.h

```diff
@@ -79,6 +80,11 @@ class LoadWorker : public IWorker {
 
   void SendInferRequest(uint32_t ctx_id, bool delayed = false)
   {
+    if (early_exit || !thread_stat_->status_.IsOk() ||
```

Comment:
Can you use the new `ShouldExit()` function here?

Reply:
yes. done

Reply:
Also, note that this was a bug that was caught by the new tests! It was possible to get here with a bad status. Before this fix we could deadlock in the request_rate worker if the backend start_stream function returned a bad status.